agenda/presentation/src/main/java/com/gkp/agenda/presentation/AgendaUiState.kt

```diff
@@ -0,0 +1,35 @@
+package com.gkp.agenda.presentation
+
+import com.gkp.agenda.presentation.util.DateWithSelected
+import com.gkp.agenda.presentation.util.getCurrentAndNextSixWeekDayDates
+import com.gkp.agenda.presentation.util.getFullNameInitials
+import java.time.LocalDate
+
+data class AgendaUiState(
+    val fullName: String = "",
+    val selectedDate: LocalDate = LocalDate.now()
+) {
+    val fullNameInitials: String
+        get() =  getFullNameInitials(fullName)
+
+    val uiDates: List<DateWithSelected>
+    get() = getCurrentAndNextSixWeekDayDates(selectedDate)
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as AgendaUiState
+
+        if (fullName != other.fullName) return false
+        if (selectedDate != other.selectedDate) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = fullName.hashCode()
+        result = 31 * result + selectedDate.hashCode()
+        return result
+    }
```

Comment:
Why are these needed here? Isn't `String` and `LocalDateTime` compared by value by default? I'd be careful with overriding equals and hashCode for states, since as soon as you add another field to this class and forget considering it in equals and hashCode, those functions will break. 99% of state classes should be able to use the default data class implementations of these.

Reply:
I removed equals and hashCode. Before the compiler was complaining